tests/py/unit/test_build_steps.py

```
@@ -0,0 +1,1078 @@
+        assert ctx.node_modules is None
+
+
+class TestBuildStep:
+    """Tests for BuildStep abstract base class."""
+
+    def test_is_abstract_class(self) -> None:
+        """BuildStep cannot be instantiated directly."""
+        with pytest.raises(TypeError):
+            BuildStep()  # type: ignore[abstract]
+
+    def test_requires_name_property(self, tmp_path: Path) -> None:
```

Ruff (0.14.13)

Unused method argument: `tmp_path`

(ARG002)